### ISSUE TITLE
use custom Qt widgets in designer

### ISSUE THREAD

**Issue by prjemian** [BCDA-APS/gemviz#133]
It would be nice if ...

... the custom widgets defined here (such as [`DateTimeRangeSlider`](https://github.com/BCDA-APS/gemviz/blob/main/gemviz/date_time_range_slider.py) or [`RangeSlider`](https://github.com/BCDA-APS/gemviz/blob/main/gemviz/range_slider.py)) by Python code could be used in other widgets in the Qt Designer.

**Comment by prjemian** [comment]
The PyDM package has a [factory method](https://github.com/slaclab/pydm/blob/ac354b744d358e05d6e0606acc114a3a511d7e5d/pydm/widgets/qtplugin_base.py#L52C5-L52C21) to assist in doing the same with the [PyDM custom widgets](https://github.com/slaclab/pydm/blob/master/pydm/widgets/qtplugins.py).

**Comment by prjemian** [comment]
Tried this in `date_time_range_slider.ui`, for `RangeSlider`.  After clicking the **Add** button, the **Promote** became enabled.  This actually worked!

**Comment by prjemian** [comment]
The `range_slider.h` file is fictitious.  Not sure why it did not raise a `FileNotFound` exception.

**Comment by prjemian** [comment]
With this change, these lines are no longer needed:

https://github.com/BCDA-APS/gemviz/blob/a9a88c27e891d572fa465d8fb23b9e89cb0cb0e8/gemviz/date_time_range_slider.py#L9

https://github.com/BCDA-APS/gemviz/blob/a9a88c27e891d572fa465d8fb23b9e89cb0cb0e8/gemviz/date_time_range_slider.py#L81-L85

**Comment by prjemian** [comment]
Widget promotion is the correct choice in this case, since these widgets are used only in our own code.  The methods used by the PyDM package apply to widgets from the package that will be used by other code, not the case here.

**Comment by prjemian** [comment]
Will wait for #128 before starting this work.
